packages/@react-spectrum/buttongroup/src/useHasOverflow.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {useEffect, useState} from 'react';
+import {useProvider} from '@react-spectrum/provider';
+import {useWindowWidth} from '@react-spectrum/utils';
+
+interface UseHasOverflowOptions {
+  skip: boolean
+}
+
+export function useHasOverflow(containerRef, options?: UseHasOverflowOptions) {
+  let {scale} = useProvider();
+  let windowWidth = useWindowWidth({debounce: 150});
+  let [hasOverflow, setHasOverflow] = useState(false);
+  let [totalChildWidth, setTotalChildWidth] = useState(0);
+
+  // Only recalculate the size of the children if the scale changes.
```

Comment:
font / locale might be grounds for triggering this as well?
I think we also need to calculate if the children themselves change? breadcrumbs get added/removed, dialog buttons sometimes get wait spinners added to them